libs/blog-bff/newsletter/src/lib/api.ts

```diff
@@ -0,0 +1,43 @@
+import { Hono } from 'hono';
+import { csrf } from 'hono/csrf';
+import { z } from 'zod';
+
+import { wpNewsletterClientMw } from '@angular-love/util-wp';
+
+const app = new Hono();
+
+app.use(
+  '*',
+  csrf({
+    origin: (origin, c) =>
+      c.env['IS_PROD'] === 'true'
+        ? /https:\/\/(\w+\.)?(angular\.love|angular-love-client\.pages\.dev)$/.test(
+            origin,
+          )
+        : true,
+  }),
+);
+
+const emailSchema = z.string().email({ message: 'Invalid email address' });
+
+app.post(
+  '/',
+  wpNewsletterClientMw,
+
+  async (c) => {
+    const newSubscriber = await c.req.text();
+
+    try {
```

Comment:
By doing `try..catch` like that you are not sure
- if the parsing of the email went wrong `const parsedNewSubscriber = emailSchema.parse(newSubscriber)`
- if the request went wrong `await c.var.wpClientSubscriber.post({...})`
- or if calling `c.json(res)` threw an exception

Reply:
@marcin-hoa I corrected it. What do you think about it ?